article/app/model/KeyEventData.scala

```diff
@@ -7,22 +7,23 @@ import com.github.nscala_time.time.Imports._
 object KeyEventData {
 
   // just for convenience for use from the templates
-  def apply(maybeBlocks: Option[Blocks], timezone: DateTimeZone, filterKeyEvents: Boolean): Seq[KeyEventData] = {
-
-    val blocks =
-      maybeBlocks.toSeq.flatMap(blocks => blocks.requestedBodyBlocks.getOrElse(CanonicalLiveBlog.timeline, blocks.body))
-
-    apply(blocks, timezone, filterKeyEvents)
+  def apply(maybeBlocks: Option[Blocks], timezone: DateTimeZone): Seq[KeyEventData] = {
+    val blocks = maybeBlocks.toSeq.flatMap(blocks => {
+      if (blocks.body.length > 0) {
+        blocks.body.filter(block => block.eventType == SummaryEvent || block.eventType == KeyEvent)
```

Comment:
I'm just asking for my own understanding. Why would we need to use `blocks.body` here? Why could we not just go with `blocks.requestedBodyBlocks` ?

Reply:
Sure, it depends on the route into this method. If we are reloading the page, the blocks come through on the body field. However, if we've polled for new blocks, they come through on the requestedBodyBlocks field. It's only ever one or the other.